hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/rollout/rollout/RolloutListGrid.java

```diff
@@ -0,0 +1,568 @@
+import org.vaadin.addons.lazyquerycontainer.BeanQueryFactory;
+import org.vaadin.addons.lazyquerycontainer.LazyQueryContainer;
+import org.vaadin.addons.lazyquerycontainer.LazyQueryDefinition;
+import org.vaadin.peter.contextmenu.ContextMenu;
+import org.vaadin.peter.contextmenu.ContextMenu.ContextMenuItem;
+import org.vaadin.peter.contextmenu.ContextMenu.ContextMenuItemClickEvent;
+import org.vaadin.spring.events.EventBus;
+import org.vaadin.spring.events.EventScope;
+import org.vaadin.spring.events.annotation.EventBusListenerMethod;
+
+import com.vaadin.data.Container;
+import com.vaadin.data.Item;
+import com.vaadin.data.util.PropertyValueGenerator;
+import com.vaadin.data.util.converter.Converter;
+import com.vaadin.server.AbstractClientConnector;
+import com.vaadin.server.FontAwesome;
+import com.vaadin.spring.annotation.SpringComponent;
+import com.vaadin.spring.annotation.ViewScope;
+import com.vaadin.ui.UI;
+import com.vaadin.ui.Window;
+import com.vaadin.ui.renderers.ClickableRenderer.RendererClickEvent;
+import com.vaadin.ui.renderers.HtmlRenderer;
+
+@SpringComponent
+@ViewScope
+public class RolloutListGrid extends AbstractGrid {
+
+    private static final long serialVersionUID = 4060904914954370524L;
+
+    private static final String UPDATE_OPTION = "Update";
+
+    private static final String RESUME_OPTION = "Resume";
+
+    private static final String PAUSE_OPTION = "Pause";
+
+    private static final String START_OPTION = "Start";
+
+    @Autowired
+    private I18N i18n;
+
+    @Autowired
+    private transient EventBus.SessionEventBus eventBus;
+
+    @Autowired
+    private transient RolloutManagement rolloutManagement;
+
+    @Autowired
+    private AddUpdateRolloutWindowLayout addUpdateRolloutWindow;
+
+    @Autowired
+    private UINotification uiNotification;
+
+    @Autowired
+    private transient RolloutUIState rolloutUIState;
+
+    @Autowired
+    private transient SpPermissionChecker permissionChecker;
+
+    @Override
+    @PostConstruct
+    protected void init() {
+        super.init();
+        eventBus.subscribe(this);
+    }
+
+    @PreDestroy
+    void destroy() {
+        eventBus.unsubscribe(this);
+    }
+
+    @EventBusListenerMethod(scope = EventScope.SESSION)
+    void onEvent(final RolloutEvent event) {
+        if (event == RolloutEvent.FILTER_BY_TEXT || event == RolloutEvent.CREATE_ROLLOUT
+                || event == RolloutEvent.UPDATE_ROLLOUT || event == RolloutEvent.SHOW_ROLLOUTS) {
+            refreshGrid();
+        }
+    }
+
+    /**
+     * Handles the RolloutChangeEvent to refresh the item in the grid.
+     * 
+     * @param rolloutChangeEvent
+     *            the event which contains the rollout which has been changed
+     */
+    @SuppressWarnings("unchecked")
+    @EventBusListenerMethod(scope = EventScope.SESSION)
+    public void onEvent(final RolloutChangeEvent rolloutChangeEvent) {
+        if (rolloutUIState.isShowRollOuts()) {
```

Comment:
early return for better readability:
 if (!rolloutUIState.isShowRollOuts()) {
return
}